randovania/patching/prime/asset_conversion.py

```diff
@@ -299,11 +299,27 @@ def id_generator(asset_type):
                 for asset in converter.converted_assets.values()
             ],
         }, meta_out, indent=4)
-
+    
+    unique_anim = []
     for asset in converter.converted_assets.values():
+        assetdata = format_for(asset.type).build(asset.resource, target_game=Game.PRIME)
+        if len(assetdata) % 32 != 0:
+            assetdata += b"\xFF" * (32-(len(assetdata) % 32))
         Path("converted").joinpath(f"{asset.id}.{asset.type.upper()}").write_bytes(
-            format_for(asset.type).build(asset.resource, target_game=Game.PRIME)
+            assetdata
         )
+        if asset.type == "ANIM":
+            for assetb in converter.converted_assets.values():
+                if assetb.type == "ANCS" and any(asset.id in x for x in converted_dependencies[assetb.id]):
+                    for dep in converted_dependencies[assetb.id]:
+                        if asset.id in unique_anim:
+                            continue
+                        if dep.type == "EVNT":
+                            with open(Path("converted").joinpath(f"{asset.id}.{asset.type.upper()}"),"r+b") as anim:
```

Comment:
Instead of hacking up the files after they're written, you could do a for loop _before_ they're written and edit there, while they're still in the decoded dict format so no magic needed.

Reply:
Fixed.